tests/python/test_utils.py

```diff
@@ -2,6 +2,26 @@
 import numpy as np
 
 
+def intersection(X, Y=None):
+    if Y is None:
+        return list(X.keys())
+    else:
+        return list(set(list(X.keys())+list(Y.keys())))
```

Comment:
Do you need to create a new list? I would hope that `np.array.__getitem__` takes any iterable

Reply:
X.keys() is a weird non python default object coming from pybind so I convert it to a list.

Reply:
I was talking of the outer `list` call. The set object should work too for indexing the numpy array.

And if `X.keys()` is iterable, you can do something like 

```py
result = set(X.keys())
if Y is not None:
    result.intersection_update(Y.Keys())
return result
```

Reply:
I don't want result to be a set that's just it and for these simple things I prefer one liners.

Reply:
This is how one could do this things and I agree there is a myriad of ways to do it. i just don't see why it matters here.

Reply:
One more thing: It looks like this is computing the union of the two sets, not their intersection. Surely we'd want the latter, no?

Reply:
> This is how one could do this things and I agree there is a myriad of ways to do it. i just don't see why it matters here.

I agree it should not matter too much performance-wise, I just like to make the code as simple as possible, potentially to avoid the kind of bug that @max-veit pointed out (which I did not see ...) 😃

Reply:
right it did not fail because more checks are in place latter. thanks for catching this Max.